django/db/migrations/state.py

```diff
@@ -135,6 +144,16 @@ def rename_model(self, app_label, old_name, new_name):
                 to_reload.add((model_state.app_label, model_state.name_lower))
         # Reload models related to old model before removing the old model.
         self.reload_models(to_reload, delay=True)
+        old_name_key = (app_label, old_name.lower())
+        new_name_key = (app_label, new_name.lower())
+        if (app_label, old_name.lower()) in self.relations:
```

Comment:
This code should only be performed if `relations` are already built.

```suggestion
        if old_name_key in self.relations:
```